service/storage_service.cc

```diff
@@ -4576,7 +4570,7 @@ future<> storage_service::on_change(gms::inet_address endpoint, const gms::appli
     // overwrites the IP back to its old value.
     // In essence, the code under the 'if' should fire if the given IP is a normal_token_owner.
     if (ep && *ep == endpoint && tm.is_normal_token_owner(host_id)) {
-        if (!is_me(endpoint)) {
+        if (!is_me(endpoint) && !_raft_topology_change_enabled) {
```

Comment:
This looks new though

Wasn't the `update_peer_info` below responsible for persisting IPs before your large PR?